### PR TITLE
Add Widgets documentation

### LINKED ISSUE #333: Widgets
Similar to how Hooks and Content Elements etc. are explained.

### PR DESCRIPTION
**Pull request by fritzmg**
Closes #333 

@xchs may be you can look over it too (to fix my punctuation 😁)

### CODE REVIEW

File: docs/dev/framework/widgets.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+* `$blnForAttribute`: controls whether a `for` attribute should be used for the widget's label.
+* `$strTemplate`: the Contao template for the widget.
+* `$strPrefix`: controls the CSS class prefix for front end widgets.
+
+
+## Creating Back End Widgets
+
+The following example creates a simple text input widget for the back end. The input
+is allowed to be submitted and the generated label should contain the `for` attribute.
+The template is set to `be_template`, which most back end widgets commonly use. It
```
Comment (fiedsch) [inline docs/dev/framework/widgets.md]: This should be `be_widget` (which is used in line 50).

File: docs/dev/framework/widgets.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+or the `$GLOBALS['TL_FFL']` array (front end widgets). 
+
+Also in both cases, the abstract method `generate()` needs to be implemented. However,
+when the widget is rendered, Contao will execute the `parse()` method of the widget
+(also both in the front and back end). It is then up to each widget's implementation
+how the output is delivered.
+
+The abstract class also contains some member variables that control the output and
+behavior of the widget. The default value of these member variables will need to be
+set in the custom child class accordingly. Typically the following member variables
```
Comment (xchs) [inline docs/dev/framework/widgets.md]:
```suggestion
set in the custom child class accordingly. Typically, the following member variables
```

File: docs/dev/framework/widgets.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            $this->name,
+            $this->id,
+            ($this->class ? ' ' . $this->class : ''),
+            StringUtil::specialchars($this->value)
+        );
+    }
+}
+```
+
+To get a better idea on the implementation details and possibilites for back end
```
Comment (xchs) [inline docs/dev/framework/widgets.md]:
```suggestion
To get a better idea on the implementation details and possibilities for back end
```

File: docs/dev/framework/widgets.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+$GLOBALS['BE_FFL']['custom_widget'] = \App\Contao\BackendWidget\CustomWidget::class;
+```
+
+Now the widget can be used as an `inputType` in your DCA.
+
+
+## Creating Front End Widgets
+
+Similar to the back end example, the following example creates a simple text input
+widget (form field) fro the form generator. The input is allowed to be submitted
```
Comment (xchs) [inline docs/dev/framework/widgets.md]:
```suggestion
widget (form field) for the form generator. The input is allowed to be submitted
```

File: docs/dev/framework/widgets.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    type="text" 
+    name="<?= $this->name ?>" 
+    id="ctrl_<?= $this->id ?>" 
+    class="text<?php if ($this->class): ?> <?= $this->class ?><?php endif; ?>" 
+    value="<?= Contao\StringUtil::specialchars($this->value) ?>"
+  >
+<?php $this->endblock(); ?>
+```
+
+To get a better idea on the implementation details and possibilites for front end
```
Comment (xchs) [inline docs/dev/framework/widgets.md]:
```suggestion
To get a better idea on the implementation details and possibilities for front end
```

**Review by xchs — COMMENTED** [review]
Just a few little typos. Otherwise, everything's fine! 👍

**Review by xchs — APPROVED** [review]

**Pull request merged**